src/clj/puppetlabs/services/request_handler/request_handler_service.clj

```diff
@@ -22,12 +32,16 @@
           (log/warn (format "%s %s"
                             (i18n/trs "The ''master.allow-header-cert-info'' setting is deprecated and will be ignored in favor of the ''authorization.allow-header-cert-info'' setting because the ''jruby-puppet.use-legacy-auth-conf'' setting is ''false''.")
                             (i18n/trs "Remove the ''master.allow-header-cert-info'' setting.")))))
-      (assoc context :request-handler
-                     (request-handler-core/build-request-handler
-                      jruby-service
-                      (request-handler-core/config->request-handler-settings
-                       config)
-                      current-code-id))))
+      ;; TODO: Throw error when max-active-requests > 0, but metrics-service is nil
+      ;; because :JRubyMetricsService isn't in bootstrap.cfg.
+      (assoc context :request-handler (if (> max-active-requests 0)
```

Comment:
Ah. I think you're thinking of `max-requests-per-instance` which is the number of requests an individual JRuby can handle before being recycled. The `max-active-requests` here is a new setting that gives the limit on outstanding requests for JRuby instances.

Would love to hear suggestions for better or less confusing names for this setting.